### PR TITLE
fix(jellyfinapi): refactors jellyfin library sync to support automatic grouping and collections

### LINKED ISSUE #256: support for automatic grouping on libraries
### Description

Current behaviour of jellyseerr is that it does not detect any libraries when automatic grouping of the libraries are enabled.

### Desired Behavior

Find a way to detect and sync with libraries with automatic grouping is enabled on jellyfin

### Additional Context

Follow up on #255 

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### LINKED ISSUE #473: Movies in collections not being detected
I have 3 libraries, TV shows, Movies and Collections but no movies that are in collections are being marked as available. Should Collections be a library item at all? It's not very clear. I've removed Collections to see what the results are after a scan

_Originally posted by @colfin22 in https://github.com/Fallenbagel/jellyseerr/discussions/472_

### LINKED ISSUE #474: Movies in collections not being detected
### Description

I have 3 libraries, TV shows, Movies and Collections but no movies that are in collections are being marked as available. Should Collections be a library item at all? It's not very clear. I've removed Collections to see what the results are after a scan

Originally posted by @colfin22 in https://github.com/Fallenbagel/jellyseerr/discussions/472

### Version

1.6.0

### Steps to Reproduce

I have 3 libraries, TV shows, Movies and Collections but no movies that are in collections are being marked as available. I've un-selected the collections library and forced a scan but the issue still persists. See uploaded screenshot, all of these movies are in my library but as they are in a collection none are marked as available. Logs of the last scan with collections enabled attached, you can see from the scan it's not detecting any of the movies in the Fast collection

### Screenshots

Screenshot 2023-09-10 113454

### Logs

```shell
2023-09-10T10:57:01.790Z [debug][Jellyfin Sync]: Title already exists and no new media types found A Fall from Grace 
2023-09-10T10:57:01.791Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fahrenheit 451 
2023-09-10T10:57:01.795Z [debug][Jellyfin Sync]: Title already exists and no new media types found False Positive 
2023-09-10T10:57:01.804Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Eyes of Tammy Faye 
2023-09-10T10:57:01.808Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fahrenheit 451 
2023-09-10T10:57:01.810Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fall 
2023-09-10T10:57:01.813Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fame 
2023-09-10T10:57:01.815Z [debug][Jellyfin Sync]: Title already exists and no new media types found Falling Down 
2023-09-10T10:57:01.820Z [debug][Jellyfin Sync]: Title already exists and no new media types found Falling Inn Love 
2023-09-10T10:57:05.988Z [debug][Jellyfin Sync]: Title already exists and no new media types found Famous and Fatal 
2023-09-10T10:57:05.992Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Family Stone 
2023-09-10T10:57:05.995Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Family I Had 
2023-09-10T10:57:05.995Z [debug][Jellyfin Sync]: Title already exists and no new media types found Far from the Madding Crowd 
2023-09-10T10:57:06.017Z [debug][Jellyfin Sync]: Title already exists and no new media types found Family Man, Family Murderer: An ID Murder Mystery 
2023-09-10T10:57:06.036Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fargo 
2023-09-10T10:57:06.038Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Farewell 
2023-09-10T10:57:06.056Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fantastic Four 
2023-09-10T10:57:06.061Z [debug][Jellyfin Sync]: Title already exists and no new media types found Family Guy Presents: It's a Trap! 
2023-09-10T10:57:06.063Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fantastic Voyage 
2023-09-10T10:57:06.064Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fanny and Alexander 
2023-09-10T10:57:06.072Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fantastic Planet 
2023-09-10T10:57:06.074Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fantasy Island 
2023-09-10T10:57:06.074Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Fanatic 
2023-09-10T10:57:06.074Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fast & Furious Presents: Hobbs & Shaw 
2023-09-10T10:57:06.111Z [debug][Jellyfin Sync]: Title already exists and no new media types found Far from the Tree 
2023-09-10T10:57:10.229Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fatal Affair 
2023-09-10T10:57:10.324Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fatale 
2023-09-10T10:57:10.332Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father of the Bride 
2023-09-10T10:57:10.333Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fear and Desire 
2023-09-10T10:57:10.333Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father Soldier Son 
2023-09-10T10:57:10.335Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Father 
2023-09-10T10:57:10.335Z [debug][Jellyfin Sync]: Title already exists and no new media types found Faults 
2023-09-10T10:57:10.336Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Fault in Our Stars 
2023-09-10T10:57:10.344Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fast Times at Ridgemont High 
2023-09-10T10:57:10.344Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father Stu 
2023-09-10T10:57:10.345Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fear and Loathing in Aspen 
2023-09-10T10:57:10.346Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fatherhood 
2023-09-10T10:57:10.354Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father Christmas Is Back 
2023-09-10T10:57:10.361Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father of the Year 
2023-09-10T10:57:10.363Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Favourite 
2023-09-10T10:57:10.369Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fatman 
2023-09-10T10:57:10.372Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fatima 
2023-09-10T10:57:10.373Z [debug][Jellyfin Sync]: Title already exists and no new media types found Father of the Bride 
2023-09-10T10:57:10.374Z [debug][Jellyfin Sync]: Title already exists and no new media types found Faster 
2023-09-10T10:57:10.374Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fast Color 
2023-09-10T10:57:14.508Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Festival 
2023-09-10T10:57:14.510Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Fifth Element 
2023-09-10T10:57:14.511Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fight Club 
2023-09-10T10:57:14.516Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Field 
2023-09-10T10:57:14.519Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fences 
2023-09-10T10:57:14.530Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Field 
2023-09-10T10:57:14.530Z [debug][Jellyfin Sync]: Title already exists and no new media types found The Fear of 13 
2023-09-10T10:57:14.531Z [debug][Jellyfin Sync]: Title already exists and no new media types found Ferris Bueller's Day Off 
2023-09-10T10:57:14.537Z [debug][Jellyfin Sync]: Title already exists and no new media types found Ferry 
2023-09-10T10:57:14.548Z [debug][Jellyfin Sync]: Title already exists and no new media types found Ferdinand 
2023-09-10T10:57:14.551Z [debug][Jellyfin Sync]: Title already exists and no new media types found Filth 
2023-09-10T10:57:14.551Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fear of Rain 
2023-09-10T10:57:14.562Z [debug][Jellyfin Sync]: Title already exists and no new media types found Feel the Beat 
2023-09-10T10:57:14.565Z [debug][Jellyfin Sync]: Title already exists and no new media types found A Few Good Men 
2023-09-10T10:57:14.565Z [debug][Jellyfin Sync]: Title already exists and no new media types found Field of Dreams 
2023-09-10T10:57:14.566Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fighting with My Family 
2023-09-10T10:57:14.567Z [debug][Jellyfin Sync]: Title already exists and no new media types found Fear and Loathing in Las Vegas 
2023-09-10T10:57:14.650Z [debug][Jellyfin Sync]: Title already exists and no new media types found Feed
```

### Platform

desktop

### Device

Docker

### Operating System

Ubuntu 22.04 host

### Browser

Firefox

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### LINKED ISSUE #515: Jellyfin: Movies grouped in a collection not synced
### Description

Jellyseerr doesn't recognize movies grouped into collections in the Jellyfin 'Movie' library.

### Version

1.7.0

### Steps to Reproduce

1. Go to Jellyfin Admin -> Libraries -> Movies -> Manage Library
2. Activate checkbox 'Automatically add to collection' (adds movies to collections, e.g. 'Iron Man Collection' so only the collection is shown)
3. Sync the movie library. Every movie grouped in a collection is missing.
4. Activate 'Collection' library to Jellyseerr (Settings -> Jellyfin -> Sync Libraries -> Activate checkbox 'Collections')
5. Jellyseerr recognizes every collection item: 'Syncing x of 209'
6. Every movie grouped in a collection is still missing in Jellyseerr. 

### Screenshots

Screenshot_20231028_204332
Screenshot_20231028_204537

### Logs

```shell
jellyseerr  | 2023-10-28T18:15:00.010Z [info][Jellyfin Sync]: Beginning to process recently added for library: Collections
jellyseerr  | 2023-10-28T18:15:00.013Z [info][Jellyfin Sync]: Recently Added Scan Complete
jellyseerr  | 2023-10-28T18:15:02.173Z [info][Jellyfin Sync]: Jellyfin Sync Starting {"sessionId":"89dd05c8-9ed2-465e-a64d-f9b087c72091"}
jellyseerr  | 2023-10-28T18:15:02.178Z [info][Jellyfin Sync]: Beginning to process library: Collections
jellyseerr  | 2023-10-28T18:15:26.504Z [info][Jellyfin Sync]: Full Scan Complete
```

### Platform

desktop

### Device

every

### Operating System

Debian Linux

### Browser

Chrome

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by fallenbagel**
#### Description
Although, #450 added support for automatic library grouping, some users reported that they were getting a 401 error when using custom authentication such as LDAP (#489). Therefore, that PR was reverted (#524).

This PR adds back the support for automatic library grouping for jellyfin authentication users (as well as some custom authentication users) using the endpoint `/Library/MediaFolders` and falls back to `/Users/${user.id}/Views` endpoint if they're unable to sync the libraries using the Media Folder endpoint (Not all LDAP users had issues. Some reported that it worked despite having custom authentication).

Once it falls back to user views endpoint for syncing, it will now detect if automatic grouping is enabled and then give a warning toast that its not supported when using some custom authentication methods. If there are no libraries, it will now show an error toast that no libraries were detected.

In the process of refactoring the library endpoints, this PR also fixed collection syncing. Now, the collection library is ignored, however, any media in a collection should sync from its respective libraries.

#### To-Dos

- [ ] Successful build `yarn build`
- [x] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #256
- Fixes #515
- Fixes #474 
- Fixes #473

### CODE REVIEW

File: src/i18n/locale/en.json
Comment (Jumail) [inline src/i18n/locale/en.json:1]: Try to keep consistency in naming. I think these lines can also be as follows:

```
"components.settings.jellyfinSyncFailedNoLibrariesFound": "No libraries were found",
"components.settings.jellyfinSyncFailedAutomaticGroupedFolders": "Custom authentication with Automatic Library Grouping not supported",
"components.settings.jellyfinSyncFailedGenericError": "Something went wrong while syncing libraries",
```

or whatever is suitable here.

File: server/api/jellyfin.ts [hunk 1]
```diff
@@ -175,46 +185,58 @@ class JellyfinAPI {
...
-      );
+      const mediaFolders = await this.axios.get<any>(`/Library/MediaFolders`);
+
+      return this.mapLibraries(mediaFolders.data.Items);
+    } catch (mediaFoldersError) {
+      // fallback to user views to get libraries
+      // this only affects LDAP users
+      try {
+        const mediaFolders = await this.axios.get<any>(
+          `/Users/${this.userId}/Views`
```
Comment (Jumail) [inline server/api/jellyfin.ts]: If ```this.userId``` is nullish shouldn't ```Me``` be used here?

File: server/routes/settings/index.ts [hunk 2]
```diff
@@ -281,6 +281,21 @@ settingsRoutes.get('/jellyfin/library', async (req, res) => {
 
     const libraries = await jellyfinClient.getLibraries();
 
+    if (libraries.length === 0) {
+      // Check if no libraries are found due to the fallback to user views
+      // This only affects LDAP users
+      const account = await jellyfinClient.getUser();
+
+      console.log(account.Configuration.GroupedFolders.length);
```
Comment (Jumail) [inline server/routes/settings/index.ts]: console log needs to be removed

**Review by Jumail — CHANGES_REQUESTED** [review]
you can update ```src/i18n/locale/en.json```. Everything else looks good.

File: server/routes/settings/index.ts (hunk 2 above)
Comment (fallenbagel) [reply server/routes/settings/index.ts]: Oh no. I didn't realise

File: server/api/jellyfin.ts (hunk 1 above)
Comment (fallenbagel) [reply server/api/jellyfin.ts]: Oh yeah true

**Pull request merged**